HW1_classes/not_stepik_home_work.py

```diff
@@ -0,0 +1,259 @@
+from Bio.Seq import Seq
+from Bio import SeqIO
+from Bio import SeqUtils
+import pylab
+
+
+# -----------------------------------------------------------------------------
+# Task 1 - simple class
+class DeadFishException(Exception):
+    pass
+
+
+class Fish:
+    def __init__(self, age, size, weight):
+        self.age = age
+        self.size = size
+        self.weight = weight
+
+    def __repr__(self):
+        s = self
+        return f'Fish: age={s.age}, size={s.size}, weight={s.weight}'
+
+    def grow(self, size=0.1, weight=0.1):
+        if not self.is_alive():
+            raise DeadFishException
+        self.age += 1
+        self.size += size
+        self.weight += weight
+
+    def kill(self):
+        self.age = -1
+
+    def is_alive(self):
+        return self.age >= 0
+
+
+def task_1_test():
+    fish = Fish(3, 4, 5)
+    print(fish)
+    fish.grow()
+    print(fish)
+    fish.kill()
+    print(fish)
+    try:
+        fish.grow()
+        print(fish)
+    except DeadFishException:
+        print("Fish can't grow - fish is dead!")
+
+
+# -----------------------------------------------------------------------------
+# Task 2 - RNA representation
+class Rna:
+    def __init__(self, seq):
+        self.rna = Seq(seq)
+
+    def translate_to_protein(self):
+        return str(self.rna.translate())
+
+    def back_transcription_to_dna(self):
+        return str(self.rna.back_transcribe())
+
+    def __repr__(self):
+        return str(self.rna)
+
+
+def task_2_test():
+    rna = Rna('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG')
+    print('RNA: ', rna)
+    print('PRO: ', rna.translate_to_protein())
+    print('DNA: ', rna.back_transcription_to_dna())
+
+
+# -----------------------------------------------------------------------------
+# Task 3 - child of Set, bad formulation of task :-(
+class PositiveSet(set):
+    # requirement:  contents only positive numbers (numbers!)
+    def __init__(self, container):
+        super().__init__()
+        for i in container:
+            if (type(i) == int or type(i) == float) and i > 0:
+                super().add(i)
+
+    # requirements:  don't add not positive elements (elements!)
+    def add(self, val):
+        try:
+            if not val > 0:
+                return
+        except Exception:
+            return  # if impossible to determine is val positive - do nothing
```

Comment:
Надо всегда указывать возможные типы ошибок. Хотя в данном случае можно было бы и не перехватывать вовсе, а то  пользователь не унает что произошло.
```suggestion
        except TypeError:
            return  # if impossible to determine is val positive - do nothing
```